tests/common.py

```diff
@@ -234,18 +235,19 @@ def __init__(self, domain=None, dependencies=None, setup=None,
         self.DOMAIN = domain
         self.DEPENDENCIES = dependencies or []
         self.REQUIREMENTS = requirements or []
+        self._setup = setup
 
         if config_schema is not None:
             self.CONFIG_SCHEMA = config_schema
 
         if platform_schema is not None:
             self.PLATFORM_SCHEMA = platform_schema
 
-        # Setup a mock setup if none given.
-        if setup is None:
-            self.setup = lambda hass, config: True
-        else:
-            self.setup = setup
+    def setup(self, hass, config):
+        """Setup the component."""
+        if self._setup is not None:
+            return self._setup(hass, config)
+        return True
```

Comment:
That is needed since a mock setup will give true on `asyncio.iscoroutine` and could not run with `run_in_executor`...